Better coroutine error handling

fraidev opened the pull request:
This PR should fix the issue #39.

I added class [CoroutineUtils.cs](https://github.com/trilitech/tezos-unity-sdk/compare/coroutine-error-handling?expand=1#diff-dd3294a76ea251b645badf21571371063992da8ca5d0bcf19bfa13ca88fd77ed) that has two methods:
- Try: Just run a coroutine with an optional handler for the exception (default is just a log).
- TryWith: Execute the Try method for a specific monoBehaviour.

k-karuna commented on Runtime/Scripts/Helpers/CoroutineUtils.cs:
```diff
@@ -0,0 +1,46 @@
+            }
+            yield return current;
+        }
+        exceptionHandler?.Invoke(null);
```
Maybe we don't need exceptionHandler invokation in cases where we have no exeptions?

fraidev replied:
Yeah, it's true. Removed.

fraidev commented:
> In my opinion this solution is good, but didn't cover all cases. For example, in case of http issues we will receive exception on [this line](https://github.com/trilitech/tezos-unity-sdk/blob/a5b7a15f52372c3ab55d41481942191c0aeba27b/Runtime/Scripts/TezosAPI/HttpClient.cs#L51) and this exception will be handled by existing [CoroutineWrapper class](https://github.com/trilitech/tezos-unity-sdk/blob/a5b7a15f52372c3ab55d41481942191c0aeba27b/Runtime/Scripts/Helpers/CoroutineWrapper.cs#L63). So, my suggestion is maybe it will be better to improve CoroutineWrapper class that we can set up errorHandler action on it, and execute it instead of current `Completed?.Invoke(default);` And when we wrap with it all coroutines that runs external callbacks and returns through `yield return coroutine` statement, looks like we will cover all possible exception cases. Moreover looks like that [WrappedRequest method](https://github.com/trilitech/tezos-unity-sdk/blob/a5b7a15f52372c3ab55d41481942191c0aeba27b/Runtime/Scripts/TezosAPI/HttpClient.cs#L65) is redundant, we can use ordinary `CoroutineWrapper` instead. If you have any thoughts, feel free to write me.

I really forgot about the CoroutineWrapper and WrappedRequest usages.
And CoroutineWrapper it was doing a similar job that the new CoroutineUtils.
What do you think to use the CoroutineUtils.Try in the WrappedRequest method and delete CoroutineWrapper because there is no more usage in the codebase?
<img width="1015" alt="image" src="https://user-images.githubusercontent.com/25258368/235775562-05fef27b-821d-4b21-b98b-1ac97d5ceeff.png">

k-karuna reviewed (commented):
This solution did not cover scenarios which are build on several `yield return IEnumerator` cases. E.g. if we click on Inventory button on UI, we will call [this method](https://github.com/trilitech/tezos-unity-sdk/blob/1e243100d95d8eb3ef45686164ac2ff89f16f9cc/Samples~/Scripts/DemoExample/Core/ExampleManager.cs#LL51C25-L51C25), that will call `GetJson<T>` method later, and if we have some network issues, we will get deserealize exception [on this line](https://github.com/trilitech/tezos-unity-sdk/blob/1e243100d95d8eb3ef45686164ac2ff89f16f9cc/Runtime/Scripts/TezosAPI/HttpClient.cs#L39), even if we pass errorHandler on `CoroutineUtils.TryWith`
So, my previous suggestion was to use something like CoroutineWrappper, that can handle Exceptions inside itself and use them wherever possible.

k-karuna commented:
@fraidev what do you think?

fraidev commented:
> @fraidev what do you think?

Make sense! I reverted the last commits and added one using the `CoroutineWrapper` with error handling. 
Can you check again?

m-kus commented:
@fraidev could you please update the changelog?

k-karuna commented:
@m-kus @fraidev I've updated changelog in order not to delay this task.
